### PR TITLE
Bug 1841118: add error handling to vSphere survey

### PR DESCRIPTION
**Pull request by patrickdillon**
Adds error handling for vSphere survey questions. This is particularly noticeable for handling user interrupts (CTRL C). Without error handling the interrupts cause survey to skip to the next question and will ultimately panic when no values have been assigned to the pointers.

### CODE REVIEW

File: pkg/asset/installconfig/vsphere/vsphere.go [hunk 1]
```diff
@@ -73,7 +73,10 @@ func Platform() (*vsphere.Platform, error) {
 		return nil, err
 	}
 
-	apiVIP, ingressVIP := getVIPs()
+	apiVIP, ingressVIP, err := getVIPs()
+	if err != nil {
+		return nil, err
```
Comment (abhinavdahiya) [inline pkg/asset/installconfig/vsphere/vsphere.go:78]:
```suggestion
		return nil, errors.Wrap(err, "failed to get VIPs")
```

File: pkg/asset/installconfig/vsphere/vsphere.go [hunk 2]
```diff
@@ -94,7 +97,8 @@ func Platform() (*vsphere.Platform, error) {
 // If creating the client fails, an error is returned.
 func getClients() (*vCenterClient, error) {
 	var vcenter, username, password string
-	survey.Ask([]*survey.Question{
+
+	err := survey.Ask([]*survey.Question{
```
Comment (abhinavdahiya) [inline pkg/asset/installconfig/vsphere/vsphere.go:101]:
```suggestion
	if err := survey.Ask([]*survey.Question{}); err != nil {
	}
```

File: pkg/asset/installconfig/vsphere/vsphere.go [hunk 3]
```diff
@@ -113,8 +120,11 @@ func getClients() (*vCenterClient, error) {
 			Validate: survey.Required,
 		},
 	}, &username)
+	if err != nil {
+		return nil, err
+	}
 
-	survey.Ask([]*survey.Question{
+	err = survey.Ask([]*survey.Question{
```
Comment (abhinavdahiya) [inline pkg/asset/installconfig/vsphere/vsphere.go:127]:
```suggestion
	if err := survey.Ask([]*survey.Question{}); err != nil {
	}
```

File: pkg/asset/installconfig/vsphere/vsphere.go [hunk 4]
```diff
@@ -177,7 +190,7 @@ func getDataCenter(ctx context.Context, finder *find.Finder, client *vim25.Clien
 	sort.Strings(dataCenterChoices)
 
 	var selectedDataCenter string
-	survey.Ask([]*survey.Question{
+	err = survey.Ask([]*survey.Question{
```
Comment (abhinavdahiya) [inline pkg/asset/installconfig/vsphere/vsphere.go:193]:
```suggestion
	if err := survey.Ask([]*survey.Question{}); err != nil {
	}
```

File: pkg/asset/installconfig/vsphere/vsphere.go [hunk 5]
```diff
@@ -216,7 +232,7 @@ func getCluster(ctx context.Context, path string, finder *find.Finder, client *v
 	sort.Strings(clusterChoices)
 
 	var selectedcluster string
-	survey.Ask([]*survey.Question{
+	err = survey.Ask([]*survey.Question{
```
Comment (abhinavdahiya) [inline pkg/asset/installconfig/vsphere/vsphere.go:235]:
```suggestion
	if err := survey.Ask([]*survey.Question{}); err != nil {
	}
```

File: pkg/asset/installconfig/vsphere/vsphere.go [hunk 6]
```diff
@@ -255,7 +274,7 @@ func getDataStore(ctx context.Context, path string, finder *find.Finder, client
 	sort.Strings(dataStoreChoices)
 
 	var selectedDataStore string
-	survey.Ask([]*survey.Question{
+	err = survey.Ask([]*survey.Question{
```
Comment (abhinavdahiya) [inline pkg/asset/installconfig/vsphere/vsphere.go:277]:
```suggestion
	if err := survey.Ask([]*survey.Question{}); err != nil {
	}
```

File: pkg/asset/installconfig/vsphere/vsphere.go [hunk 7]
```diff
@@ -300,7 +323,7 @@ func getNetwork(ctx context.Context, path string, finder *find.Finder, client *v
 	sort.Strings(networkChoices)
 
 	var selectednetwork string
-	survey.Ask([]*survey.Question{
+	err = survey.Ask([]*survey.Question{
```
Comment (abhinavdahiya) [inline pkg/asset/installconfig/vsphere/vsphere.go:326]:
```suggestion
	if err := survey.Ask([]*survey.Question{}); err != nil {
	}
```

File: pkg/asset/installconfig/vsphere/vsphere.go [hunk 8]
```diff
@@ -310,14 +333,17 @@ func getNetwork(ctx context.Context, path string, finder *find.Finder, client *v
...
 
 	return selectednetwork, nil
 }
 
-func getVIPs() (string, string) {
+func getVIPs() (string, string, error) {
 	var apiVIP, ingressVIP string
 
-	survey.Ask([]*survey.Question{
+	err := survey.Ask([]*survey.Question{
```
Comment (abhinavdahiya) [inline pkg/asset/installconfig/vsphere/vsphere.go:346]:
```suggestion
	if err := survey.Ask([]*survey.Question{}); err != nil {
	}
```

File: pkg/asset/installconfig/vsphere/vsphere.go [hunk 9]
```diff
@@ -328,8 +354,11 @@ func getVIPs() (string, string) {
 			}),
 		},
 	}, &apiVIP)
+	if err != nil {
+		return "", "", err
+	}
 
-	survey.Ask([]*survey.Question{
+	err = survey.Ask([]*survey.Question{
```
Comment (abhinavdahiya) [inline pkg/asset/installconfig/vsphere/vsphere.go:361]:
```suggestion
	if err := survey.Ask([]*survey.Question{}); err != nil {
	}
```

**Comment by patrickdillon** [comment]
updated PTAL thank you!

File: pkg/asset/installconfig/vsphere/vsphere.go [hunk 10]
```diff
@@ -94,35 +97,43 @@ func Platform() (*vsphere.Platform, error) {
 // If creating the client fails, an error is returned.
 func getClients() (*vCenterClient, error) {
 	var vcenter, username, password string
-	survey.Ask([]*survey.Question{
+	var err error
+
+	if err = survey.Ask([]*survey.Question{
```
Comment (abhinavdahiya) [inline pkg/asset/installconfig/vsphere/vsphere.go:102]: no need to use an existing err, it just shadows the err and can cause over writes when not desired.

```suggestion
	var err error

	if err := survey.Ask([]*survey.Question{
```

Comment (patrickdillon) [reply pkg/asset/installconfig/vsphere/vsphere.go]: oh right! updated

**Pull request merged**